fake/src/impls/geo/mod.rs

```diff
@@ -0,0 +1,204 @@
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::Line::<T> {
+            start: Faker.fake_with_rng(rng),
+            end: Faker.fake_with_rng(rng),
+        }
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::LineString<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::LineString::<T>::new(Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::MultiLineString<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::MultiLineString::<T>::new(Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Point<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::Point::<T>::new(Faker.fake_with_rng(rng), Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::MultiPoint<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::MultiPoint::<T>::new(Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Polygon<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        // Polygon will auto-close these LineString.
+        geo_types::Polygon::<T>::new(Faker.fake_with_rng(rng), Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::MultiPolygon<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::MultiPolygon::<T>::new(Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Rect<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        // Rect points can not overlap.
+        let nums: Vec<T> = crate::unique::<T, _>(rng, 4);
+        let coord_1 = geo_types::Coord::<T> {
+            x: nums[0],
+            y: nums[1],
+        };
+        let coord_2 = geo_types::Coord::<T> {
+            x: nums[2],
+            y: nums[3],
+        };
+        geo_types::Rect::new::<geo_types::Coord<T>>(coord_1, coord_2)
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Triangle<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        // Triangle cant be a straight line, so avoid two identical slopes.
+        // There is a bug in slope that requires a different implementation
+        // https://github.com/georust/geo/issues/1001
+        // Also this uses f64 to avoid similar slopes when T is an integer.
+        fn line_slope<T>(a: geo_types::Coord<T>, b: geo_types::Coord<T>) -> f64
+        where
+            T: CoordNum,
+        {
+            let (min_x, max_x) = if a.x > b.x { (b.x, a.x) } else { (a.x, b.x) };
+            let (min_y, max_y) = if a.y > b.y { (b.y, a.y) } else { (a.y, b.y) };
```

Comment:
optional: I'm not sure what is cleaner, but rather than the if statements it may be possible to use an absolute value for slope so that the order doesn't matter.

Reply:
The order matters otherwise the minus will can overflow the integer range.